TriblerGUI/widgets/trustdisplaypage.py

```diff
@@ -0,0 +1,161 @@
+        """
+        url = "display?{0}{1}".format("focus_node={0}&".format(self.focus_node) if self.focus_node else "",
+                                      "neighbor_level={0}".format(str(self.neighbor_level)))
+        self.request_mgr.perform_request(url, self.process_display_information)
+
+    @staticmethod
+    def process_display_information(information):
+        """
+        Process the JSON dictionary in the HTTP Response for the data used in the visualization.
+
+        The JSON will be converted to the following format:
+            {
+                "focus_node": "xyz",
+                "neighbor_level: 1
+                "nodes": [{
+                    "public_key": "xyz",
+                    "page_rank": 5
+                }, ...],
+                "edges": [{
+                    "from": "xyz",
+                    "to": "xyz_n1",
+                    "amount_up": 100,
+                    "amount_down": 10,
+                    "ratio": 0.90909,
+                    "log_ratio": 0.66666
+                }, ...]
+            }
+        
+
+        :param information: the JSON dictionary passed on by the HTTP Request
+        """
+        focus_node = information["focus_node"]
+
+        grouped_edges = TrustDisplayPage.group_elements(information["edges"], "from")
+
+        # TODO: Do the procedure below for each of the nodes in the graph, whilst erasing duplicates.
+        focus_node_edges = TrustDisplayPage.get_combined_edges(grouped_edges, focus_node)
+
+        # TODO: Send the data to the display to display the data
+        return {"nodes": information["nodes"], "edges": focus_node_edges}
+
+    @staticmethod
+    def get_combined_edges(grouped_edges, node_name):
+        """
+        Combine the directed edges between the given node and other nodes to one edge per pair.
+
+        The attributes of the combined edges are calculated as follows:
+            - from: node_name
+            - to: to attribute from outgoing edge from node_name
+            - amount_up: amount from the outgoing edge from node_name
+            - amount_down: amount from the ingoing edge to node_name if any
+            - ratio: amount_up / (amount_up + amount_down)
+            - log_ratio: log(amount_up + 1) / (log(amount_up + 1) + log(amount_down + 1))
+
+        :param grouped_edges: the dictionary of edges, grouped by "from" attribute
+        :param node_name: the node name from which viewpoint each combine edge is created
+        :return: a list of combined edges with the described attributes
+        """
+        combined_edges = []
+        for edge in grouped_edges[node_name]:
+            inverse_edge = next((inv for inv in grouped_edges[edge["to"]] if inv["to"] == node_name), None)
+            up = edge["amount"]
+            down = 0
+            ratio = 0
+            log_ratio = 0
+            if inverse_edge:
+                down = inverse_edge["amount"]
```

Comment:
There are no edges which have an amount smaller or equal to zero, since that would make no sense in the whole multichain idea; it is a transport from one node to another